src/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -173,21 +175,24 @@ public static void CreateErrorResponse(HttpResponse response, RequestContext req
 
             try
             {
-                var msalTokenResponse = JsonHelper.DeserializeFromJson<MsalTokenResponse>(response.Body);
+                MsalTokenResponse msalTokenResponse = JsonHelper.DeserializeFromJson<MsalTokenResponse>(response.Body);
 
                 if (CoreErrorCodes.InvalidGrantError.Equals(msalTokenResponse.Error, StringComparison.OrdinalIgnoreCase))
                 {
                     throw MsalExceptionFactory.GetUiRequiredException(
                         CoreErrorCodes.InvalidGrantError,
                         msalTokenResponse.ErrorDescription,
                         null,
-                        ExceptionDetail.FromHttpResponse(response));
+                        response, 
+                        msalTokenResponse);
```

Comment:
>msalTokenResponse [](start = 24, length = 17)

Given we pass this in... perhaps we can simplify the params passed in since we are anyway using the value from the msalTokenCache (same below)